src/audio/dai-zephyr.c

```diff
@@ -1122,7 +1122,7 @@ static int dai_comp_trigger_internal(struct dai_data *dd, struct comp_dev *dev,
 		break;
 	case COMP_TRIGGER_PAUSE:
 		comp_dbg(dev, "dai_comp_trigger_internal(), PAUSE");
-#if CONFIG_COMP_DAI_TRIGGER_ORDER_REVERSE
+#if CONFIG_COMP_DAI_STOP_TRIGGER_ORDER_REVERSE
```

Comment:
That commit added the Kconfig:
config COMP_DAI_**STOP_**TRIGGER_ORDER_REVERSE

and in code it used:
#if CONFIG_COMP_DAI_TRIGGER_ORDER_REVERSE

This patch corrects that config option use in the code.

Reply:
@ranj063, @plbossart, the drain issue on UPX is not the same, hw:0,1 does not work at all, it always fail, but it allowed me to have a fix for the handling of such.

I still think that this PR will help the MTL issue.

Reply:
@ujfalusi can you please remind me what this kconfig CONFIG_DMA_DW_SUSPEND_DRAIN is meant for?

Reply:
@ranj063, in case you want to stop the DMA without loosing data in it's FIFO (we don't afaik) then the recommended stop sequence is:
SUSPEND+DRAIN and wait  for drain to complete
Clear the EN bit to disable the channel

In case of pause you would need only toggle the SUSPEND bit.

There is no case documented when you just want to disable the channel and you don't care about the FIFO, but it is afaik is the clearing of EN (the FIFO will be lost).

Imho the correct fix would be to _not_ enable `CONFIG_DMA_DW_SUSPEND_DRAIN` for any Intel platform.